Description: Don't run .ts via Node.js directly
Author: Daniel Richard G. <skunk@iSKUNK.ORG>

Direct execution of TypeScript only works with Node.js >= v26, or tsx(1).
Swap in a handcrafted dummy enum_conversions.ts file instead of running the
generation logic. As we disable glic anyway (see disable-ai.patch), this
should hopefully not break anything.

Also, don't run run_tsc.py / tsc.js, as it breaks with Node.js v12 and
the output is just a stamp file anyway.

--- a/chrome/browser/resources/glic/glic_api_impl/generate_impl/BUILD.gn
+++ b/chrome/browser/resources/glic/glic_api_impl/generate_impl/BUILD.gn
@@ -36,15 +36,13 @@ ts_library("check_generator_types") {
 }
 
 node("generate_enum_conversions") {
-  script = "//third_party/node/node.py"
+  script = "//debian/dummy/copy_file.py"
   inputs = [
-    "$target_gen_dir/gen_enum_conversions.js",
-    "$target_gen_dir/mojom_ast.json",
+    "//debian/dummy/enum_conversions.ts"
   ]
   outputs = [ "$target_gen_dir/enum_conversions.ts" ]
   args = [
-    rebase_path("$target_gen_dir/gen_enum_conversions.js", root_build_dir),
-    rebase_path("$target_gen_dir/mojom_ast.json", root_build_dir),
+    rebase_path(inputs[0], root_build_dir),
     rebase_path(outputs[0], root_build_dir),
   ]
   deps = [
